v2/server.go

```diff
@@ -91,6 +92,15 @@ func (s *Server) RegisterBeforeFunc(f func(i *RequestInfo)) {
 	s.beforeFunc = f
 }
 
+// RegisterValidateRequestFunc registers the specified function as the function
+// that will be called after the BeforeFunc (if registered) and before invoking
+// the actual Service method. If this function returns a non-nil error, the method
+// won't be invoked and this error will be considered as the method result.
+// The argument of this function is the already-unmarshalled *args parameter of the method.
+func (s *Server) RegisterValidateRequestFunc(f func(i interface{}) error) {
+	s.validateFunc = f
```

Comment:
If you reflect.ValueOf this here, then below you don't have to get out of Reflectworld and in again to call this function.

Reply:
Good idea, thanks!